### PR TITLE
add developer requirements to CONTRIBUTING.md

### LINKED ISSUE #3749: Failing tests on fresh checkout
When I checkout ember-cli and try to run the test suite as described in https://github.com/ember-cli/ember-cli/blob/master/CONTRIBUTING.md there are failing tests.

I am running `npm install && npm run test-all`

Test suite output here -> https://gist.github.com/jakehow/58405aad30b5295fe0b5

npm-debug.log -> https://gist.github.com/jakehow/b5a212030ad414c43cdc

This is a fresh checkout of master on Yosemite, npm@2.7.1, node@v0.10.37

### PR DESCRIPTION
**Pull request by jakehow**
This should close #3749 

If there are more specifics that might be good to explain here, let me know.  Right now it does not say phantomjs anywhere in README.md either so this was a little confusing to me.

The failing tests do point you in this direction but the messages are a little buried ( I did not see it right away).

### CODE REVIEW

**Comment by jakehow** [comment]
Seems like an unrelated failure. Is this spec intermittently failing?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -66,20 +66,25 @@ We love pull requests. Here's a quick guide:
 
 1. Fork the repo.
 
-2. Run the tests. We only take pull requests with passing tests, and it's great
+2. Ensure you have the development requirements:
+
+   * node & npm
```
Comment (stefanpenner) [inline CONTRIBUTING.md]: * we should specify the versions
* we should specifiy that these must be installed without sudo

Comment (jakehow) [inline CONTRIBUTING.md]: @stefanpenner do you know the minimum version requirements for each dependency off hand? Happy to update the PR.

Comment (rwjblue) [inline CONTRIBUTING.md]: We recommend either Node 0.12 or iojs 1.x. And npm should be 2.x.

Comment (jakehow) [inline CONTRIBUTING.md]: @stefanpenner @rwjblue updated.

**Pull request merged**